### PR TITLE
4 - Data Layer Overview & Models

### PR DESCRIPTION
**Pull request by MarcoMartins2417**
- Created NumberTrivia Model
- Created NumberTrivia Model Test
- Created Fixtures for Trivia Numbers
- Created a Fixture Reader File in order to use the Fixtures
- Created 2 New Tests in order to Test the Json Files
- Add the Method and the Test for toJson

### CODE REVIEW

File: number_trivia_app/test/features/number_trivia/data/models/number_trivia_model_test.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      },
+    );
+
+    test(
+      'should return a valid model when the JSON number is regarded as a double',
+      () async {
+        // act
+        final result = tNumberTriviaModel.toJson();
+        // assert
+        final expectedMap = {"text": "Test Text", "number": 1,};
```
Comment (andre-exd) [inline number_trivia_app/test/features/number_trivia/data/models/number_trivia_model_test.dart]: Fix formatting

**Review by andre-exd — CHANGES_REQUESTED** [review]

File: number_trivia_app/test/features/number_trivia/data/models/number_trivia_model_test.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import 'dart:convert';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:number_trivia_app/features/number_trivia/data/models/number_trivia_model.dart';
+import 'package:number_trivia_app/features/number_trivia/domain/entities/number_trivia.dart';
+
+import '../../../../fixtures/fixture_reader.dart';
+
+void main() {
+  final tNumberTriviaModel = NumberTriviaModel(number: 1, text: 'Test Text');
```
Comment (andre-exd) [inline number_trivia_app/test/features/number_trivia/data/models/number_trivia_model_test.dart:10]: Should you add the named parameters on different lines?

File: number_trivia_app/lib/features/number_trivia/data/models/number_trivia_model.dart [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import '../../domain/entities/number_trivia.dart';
+
+class NumberTriviaModel extends NumberTrivia {
+  NumberTriviaModel({
+    @required String text,
+    @required int number,
+  }) : super(text: text, number: number);
+
+  factory NumberTriviaModel.fromJson(Map<String, dynamic> json) {
+    return NumberTriviaModel(text: json['text'], number: (json['number'] as num).toInt());
```
Comment (andre-exd) [inline number_trivia_app/lib/features/number_trivia/data/models/number_trivia_model.dart:12]: Should you add the named parameters on different lines?

**Review by andre-exd — CHANGES_REQUESTED** [review]

File: number_trivia_app/test/features/number_trivia/data/models/number_trivia_model_test.dart (hunk 2 above)
Comment (MarcoMartins2417) [reply number_trivia_app/test/features/number_trivia/data/models/number_trivia_model_test.dart:10]: We discussed the topic offline

File: number_trivia_app/lib/features/number_trivia/data/models/number_trivia_model.dart (hunk 3 above)
Comment (MarcoMartins2417) [reply number_trivia_app/lib/features/number_trivia/data/models/number_trivia_model.dart:12]: We discussed the topic offline

**Review by andre-exd — APPROVED** [review]

**Pull request merged**